### PR TITLE
camera setting fov for second viewport

### PR DESCRIPTION
**Pull request by jproust**
Fixes [AB#2047](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/2047)
Also move the mapMode setting to cameraSettings

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -2564,10 +2567,9 @@ void vcMain_RenderSceneWindow(vcState *pProgramState)
         pProgramState->pActiveViewport->camera.headingPitch = udDouble2::create(0.0f, -UD_HALF_PI);
 
         // TODO: Hack! (Until we resolve multiple camera settings)
```
Comment (peterAdams1014) [inline src/main.cpp:2569]: This comment can be removed now.

File: src/main.cpp [hunk 2]
```diff
@@ -2564,10 +2567,9 @@ void vcMain_RenderSceneWindow(vcState *pProgramState)
...
-        vcCameraSettings settingsCopy = pProgramState->settings.camera;
-        settingsCopy.fieldOfView = UD_PIf * 10.0f / 18.f; // 10 degrees
+        //vcCameraSettings settingsCopy = pProgramState->settings.camera;
```
Comment (peterAdams1014) [inline src/main.cpp:2570]: Remove this.

File: src/scene/vcViewpoint.cpp [hunk 3]
```diff
@@ -249,27 +249,30 @@ void vcViewpoint::ApplySettings(vcState *pProgramState)
...
-    pProgramState->settings.camera.lensIndex = vcLS_50mm;
-  else if (udStrEqual(pStringVal, "70mm"))
-    pProgramState->settings.camera.lensIndex = vcLS_70mm;
-  else if (udStrEqual(pStringVal, "100mm"))
-    pProgramState->settings.camera.lensIndex = vcLS_100mm;
-
-  doubleVal = pProgramState->settings.camera.fieldOfView;
-  udProjectNode_GetMetadataDouble(m_pNode, "visualisation.camera_lens.custom_radians", &doubleVal, doubleVal);
-  pProgramState->settings.camera.fieldOfView = (float)doubleVal;
+  for (int v = 0; v < vcMaxViewportCount; v++) {
```
Comment (peterAdams1014) [inline src/scene/vcViewpoint.cpp:252]: Brace on newline.

File: src/vcSettings.cpp [hunk 4]
```diff
@@ -567,7 +564,13 @@ bool vcSettings_Load(vcSettings *pSettings, bool forceReset /*= false*/, vcSetti
       udStrcpy(pSettings->loginInfo.email, data.Get("login.username").AsString());
 
     // Camera
-    pSettings->camera.moveSpeed = data.Get("camera.moveSpeed").AsFloat(10.f);
+    for (int v = 0; v < vcMaxViewportCount; v++)
+    {
+      pSettings->camera.moveSpeed[v] = data.Get("camera.moveSpeed[%d]", v).AsFloat(10.f);
```
Comment (peterAdams1014) [inline src/vcSettings.cpp:569]: These all need to be backwards compatibility, use the old saved settings as input:

E.g.:
```
pSettings->camera.moveSpeed[v] = data.Get("camera.moveSpeed[%d]", v).AsFloat(data.Get("camera.lensId"));
```
and
```
pSettings->camera.mapMode[v] = data.Get("camera.mapMode[%d]", v).AsBool(data.Get("viewports.viewport[%d].mapMode", v).AsBool(false));
```

File: src/vcSettings.cpp [hunk 5]
```diff
@@ -752,11 +754,15 @@ bool vcSettings_Save(vcSettings *pSettings)
   }
 
   // Camera
-  data.Set("camera.moveSpeed = %f", pSettings->camera.moveSpeed);
+  for (int v = 0; v < vcMaxViewportCount; v++)
+  {
+    data.Set("camera.moveSpeed[%d] = %f", v, pSettings->camera.moveSpeed[v]);
```
Comment (peterAdams1014) [inline src/vcSettings.cpp:759]: Same as above, all need to be backwards compatible.

**Review by peterAdams1014 — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 1 above)
Comment (jproust) [reply src/main.cpp]: Fixed

File: src/main.cpp (hunk 2 above)
Comment (jproust) [reply src/main.cpp]: Fixed

File: src/scene/vcViewpoint.cpp (hunk 3 above)
Comment (jproust) [reply src/scene/vcViewpoint.cpp]: Fixed

File: src/vcSettings.cpp (hunk 4 above)
Comment (jproust) [reply src/vcSettings.cpp]: Fixed

File: src/vcSettings.cpp (hunk 5 above)
Comment (peterAdams1014) [reply src/vcSettings.cpp:759]: My comment is wrong. Ignore this (I thought this was another read).

**Pull request merged**